probe.go

```diff
@@ -801,6 +801,140 @@ func probeLinkMonitor(c FortiHTTP) ([]prometheus.Metric, bool) {
+		)
+		mPacketSent = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_packetsent",
+			"Number of packets sent for this health check",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+		mPacketReceived = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_packetreceived",
+			"Number of packets received for this health check",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+		mSession = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_session",
+			"Session count for the interface",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+		mTXBandwidth = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_txbandwidth",
+			"Upload bandwidth of the interface",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+		mRXBandwidth = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_rxbandwidth",
+			"Download bandwidth of the interface",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+		mStateChanged = prometheus.NewDesc(
+			"fortigate_virtualwan_healthcheck_statechanged",
+			"Timestamp of the last link state change, in seconds.",
+			[]string{"vdom", "sla", "interface"}, nil,
+		)
+	)
+
+	type SLAMember struct {
+		Status         string  `json:"status"`
+		Latency        float64 `json:"latency"`
+		Jitter         float64 `json:"jitter"`
+		PacketLoss     float64 `json:"packet_loss"`
+		PacketSent     float64 `json:"packet_sent"`
+		PacketReceived float64 `json:"packet_received"`
+		//todo add slatargetmet
+		SLAtargetmet   []string  `json:"sla_targets_met"`
+		Session        float64 `json:"session"`
+		TxBandwidth    float64 `json:"tx_bandwidth"`
+		RxBandwidth    float64 `json:"rx_bandwidth"`
+		StateChanged   float64 `json:"state_changed"`
+	}
+
+	type VirtualWanSLA map[string]SLAMember
+
+	type VirtualWanMonitorResponse struct {
+		HTTPMethod string               `json:"http_method"`
+		Results    map[string]VirtualWanSLA `json:"results"`
+		VDOM       string               `json:"vdom"`
+		Path       string               `json:"path"`
+		Name       string               `json:"name"`
+		Status     string               `json:"status"`
+		Serial     string               `json:"serial"`
+		Version    string               `json:"version"`
+		Build      int                  `json:"build"`
+	}
+
+
+	var rs []VirtualWanMonitorResponse
+
+	if err := c.Get("api/v2/monitor/virtual-wan/health-check","vdom=*", &rs); err != nil {
+		log.Printf("Error: %v", err)
+		return nil, false
+	}
+	m := []prometheus.Metric{}
+	for _, r := range rs {
+		for VirtualWanSLAName, VirtualWanSLA := range r.Results {
+			for MemberName, Member := range VirtualWanSLA {
+				wanStatusUp, wanStatusDown, wanStatusError, wanStatusUnknown := 0.0, 0.0, 0.0, 0.0
+				switch Member.Status {
+				case "up":
+					wanStatusUp = 1.0
+					break
+				case "down":
+					wanStatusDown = 1.0
+					break
+				case "error":
+					wanStatusError = 1.0
+					break
+				case "disable":
+					wanStatusError = 1.0
```

Comment:
Hmm. `disable` is not really an error if I'm not mistaken. I think we should do here a own value.

Reply:
Thanks for the feedback, it is my copy/paste mistake, added a "StatusDisable" variable.